### PR TITLE
fix: handle errors returned by `Close()` in write paths

### LINKED ISSUE #1516: `oras` should handle errors returned by `Close()` in all write paths
As mentioned in this [comment](https://github.com/oras-project/oras/pull/1509#discussion_r1796502216), `oras` should have an issue to track handling errors returned by Close() in all write paths.

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1516 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
/lgtm

Although it would read nicer if there wasn't a swtich from `fileCloseErr` to `closeErr`

File: cmd/oras/internal/display/content/manifest_fetch.go [hunk 1]
```diff
@@ -31,14 +31,19 @@ type manifestFetch struct {
 	outputPath string
 }
 
-func (h *manifestFetch) OnContentFetched(desc ocispec.Descriptor, manifest []byte) error {
+func (h *manifestFetch) OnContentFetched(desc ocispec.Descriptor, manifest []byte) (err error) {
 	out := h.stdout
 	if h.outputPath != "-" && h.outputPath != "" {
 		f, err := os.Create(h.outputPath)
```
Comment (shizhMSFT) [inline cmd/oras/internal/display/content/manifest_fetch.go:37]: It seems the `err` here will override the `err` at line 34.

Comment (wangxiaoxuan273) [reply cmd/oras/internal/display/content/manifest_fetch.go:37]: Renamed the variables.

**Comment by wangxiaoxuan273** [comment]
> /lgtm
> 
> Although it would read nicer if there wasn't a swtich from `fileCloseErr` to `closeErr`

Thanks Terry. Renamed it.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**